rllib/models/torch/visionnet.py

```diff
@@ -131,7 +131,18 @@ def __init__(
                 padding, _ = same_padding(in_size, [1, 1], [1, 1])
                 if post_fcnet_hiddens:
                     layers.append(nn.Flatten())
-                    in_size = out_channels
+                    # If we have a post-fc-stack, we can allow arbitrary shapes for the final Conv2D layer.
+                    # So we use a dummy sample to determine the size of the Conv2D output (= post-fc-stack input)
+                    self._convs = nn.Sequential(*layers)
+                    # Create a B=1 dummy sample and push it through out conv-net.
+                    dummy_in = (
+                        torch.from_numpy(self.obs_space.sample())
```

Comment:
this should probably work fine given that this `obs_space` is the always the most flattened observation space, not necessarily the original observation space IIUC.